#!/bin/sh -e
## 307_ld-pic.dpatch
##
## DP: Description: Fix failing ld-shared tests when built with gcc 4.3
## DP: Upstream status: Fix in CVS head

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p0 < $0;;
       -unpatch) patch $patch_opts -p0 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

2008-02-14  H.J. Lu  <hongjiu.lu@intel.com>

	* ld-shared/sh1.c (shlib_overriddencall2): Moved to ...
	* ld-shared/sh2.c (shlib_overriddencall2): Here.  New.

@DPATCH@
--- ld/testsuite/ld-shared/sh1.c.pic	1999-06-08 20:04:03.000000000 -0700
+++ ld/testsuite/ld-shared/sh1.c	2008-02-14 11:15:20.000000000 -0800
@@ -71,12 +71,6 @@ shlib_shlibcall2 ()
 {
   return shlib_overriddencall2 ();
 }
-
-int
-shlib_overriddencall2 ()
-{
-  return 7;
-}
 #endif
 
 /* This function calls a function defined by the main program.  */
--- ld/testsuite/ld-shared/sh2.c.pic	1999-05-03 00:29:09.000000000 -0700
+++ ld/testsuite/ld-shared/sh2.c	2008-02-14 11:15:38.000000000 -0800
@@ -12,3 +12,11 @@ shlib_shlibcalled ()
 {
   return 5;
 }
+
+#ifndef XCOFF_TEST
+int
+shlib_overriddencall2 ()
+{
+  return 7;
+}
+#endif
